components/resource_metering/src/model.rs

```diff
@@ -64,10 +65,36 @@ impl Default for RawRecords {
             begin_unix_time_secs: now_unix_time.as_secs(),
             duration: Duration::default(),
             records: HashMap::default(),
+            others: RawRecord::default(),
         }
     }
 }
 
+impl RawRecords {
+    /// Keep a maximum of `k` self.records and aggregate the others into self.others.
+    pub fn keep_top_k(&mut self, k: usize) {
+        if self.records.len() <= k {
+            return;
+        }
+        let mut buf = STATIC_BUF.with(|b| b.take());
+        buf.clear();
+        // Find kth top cpu time.
+        for record in self.records.values() {
+            buf.push(record.cpu_time);
+        }
+        pdqselect::select_by(&mut buf, k, |a, b| b.cmp(a));
```

Comment:
How about select_by self.records directly? Then you don't need to use the buf any more.

Reply:
`self.records` is an unordered `HashMap`, but `pdqselect::select_by()` requires to be done on `&mut [T]`, it will sort in place.